quupe-RN/js/screens/Calendar/Calender.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import {
+    View,
+    Text,
+    Button,
+    TouchableOpacity,
+    ActivityIndicator
+} from 'react-native';
+import Icon from 'react-native-vector-icons/Ionicons';
+import { Calendar } from 'react-native-calendars';
+import gql from 'graphql-tag';
+import { Query } from 'react-apollo';
+import Overlay from 'react-native-modal-overlay';
+import Moment from 'moment';
+import PropTypes from 'prop-types';
+import styles from './styles';
+import { formatCalendarData } from '../../lib/functions';
+
+const itemBorrowedQuery = gql`
+    query Item($id: ID!) {
+        Item(id: $id) {
+            allBorrowers {
+                startDate
+                endDate
+            }
+        }
+    }
+`;
+
+const CalendarScreen = props => (
+    <View style={styles.calendarScreen}>
+        <View style={styles.fromToContainer}>
+            <Button
+                title={
+                    props.fromDate === 'From'
+                        ? 'From'
+                        : Moment(props.fromDate).format('D MMM Y')
+                }
+                onPress={() => props.displayFromCalendar()}
+            />
+            <Button
+                title={
+                    props.toDate === 'To'
+                        ? 'To'
+                        : Moment(props.toDate).format('D MMM Y')
+                }
+                onPress={() => props.displayToCalendar()}
+            />
+        </View>
+        <Icon
+            name="ios-arrow-round-back"
+            size={35}
+            style={styles.backButton}
+            color="black"
+            onPress={() => props.navigation.pop()}
+        />
+        <Text style={styles.calendarTitle}>{props.title}</Text>
+        {props.fromCalendar && (
+            <Query query={itemBorrowedQuery} variables={{ id: props.itemId }}>
+                {({ loading, error, data }) => {
+                    if (loading) return <ActivityIndicator />;
+                    if (error) return <Text>Error</Text>;
+                    const calendarData = formatCalendarData(data.Item.allBorrowers);
+                    return (
+                        <View>
+                            <Text>From</Text>
+                            <Calendar
+                                style={styles.calendar}
+                                onDayPress={day => {
+                                    props.setFromDate(day.dateString);
+                                }}
+                                minDate={new Date()}
+                                markingType="basic"
+                                markedDates={calendarData}
+                            />
+                        </View>
+                    );
+                }}
+            </Query>
+        )}
+        {props.toCalendar && (
+            <Query query={itemBorrowedQuery} variables={{ id: props.itemId }}>
```

Comment:
Can we find another way to do this? We should not be doing the same queries twice.